### ISSUE TITLE
[BUG] BTC RPC Explorer breaks on Bitcoin Core restart

### ISSUE THREAD

**Issue by Stadicus** [raspibolt/raspibolt#848]
### Describe the bug
If Bitcoin Core is restarted, the .cookie file is recreated with new access credentials. Currently, BTC RPC Explorer does not detect this change and just keeps trying to query bitcoind with the old credentials it read from the cookie on startup. This renders the application unusable, until it is restarted (reading the new credentials and accessing bitcoind successfully again).

![image](https://user-images.githubusercontent.com/19550140/146669014-055a1bb7-4c43-4c7a-a5c7-7fa55c33b9e6.png)

### Location in guide
<https://raspibolt.org/btcrpcexplorer.html>

### Expected behavior
BTC RPC Explorer should detect a change in the `.cookie` file content and always use the latest RPC access credentials.

This is not something this guide can fix, so I opened an issue in the project's repository:  
https://github.com/janoside/btc-rpc-explorer/issues/393

**Comment by VajraOfIndra** [comment]
Should we create an Issue in the BTC RPC Explorer to ask the author how to deal with this?

**Comment by kristapsk** [comment]
I have noticed this too, some time ago.

> This is not something this guide can fix

It could - autorestart BTC RPC Explorer if restart of Bitcoin Core is detected.

> so I opened an issue in the project's repository

But agree that fixing upstream would be the best solution.

**Comment by kristapsk** [comment]
Problem still exists, don't think this should have been closed.

**Comment by Stadicus** [comment]
I asked in https://github.com/janoside/btc-rpc-explorer/issues/393 if a fix is planned

**Comment by VajraOfIndra** [comment]
> It could - autorestart BTC RPC Explorer if restart of Bitcoin Core is detected.

How could this be done?

I've looked at the service file option. I found `BindsTo=` that would allow to automatically stop the Explorer when Bitcoin Core is stopped, but I didn't find anything (so far) that allows to restart it when Bitcoin Core starts... What do you have in mind?

**Comment by VajraOfIndra** [comment]
So I looked into the systemd [Unit] options a bit more.

I assume we have the following goals:

1. Start the Explorer if Bitcoin Core starts
2. When starting Bitcoin Core, if the Explorer can't start (e.g. not installed), Bitcoin Core should still starts.
3. The explorer should start only AFTER Bitcoin Core has started
4. Stop the Explorer if Bitcoin Core stops

Also, either:
--5a. If the Explorer is manually started, do not start Bitcoin Core
Or
--5b. If the Explorer is manually started, also start Bitcoin Core

For 1,2,3,4 and 5a, we could use:

`bitcoind.service`:
```
[Unit]
After=network.target
Wants=btcrpcexplorer.service
```

`btcrpcexplorer.service`:
```
[Unit]
After=bitcoind.service
PartsOf=bitcoind.service
```

For 1,2,3,4 and 5b, we would need (`BindsTo` instead of `PartsOf`):

`bitcoind.service`:
```
[Unit]
After=network.target
Wants=btcrpcexplorer.service
```

`btcrpcexplorer.service`:
```
[Unit]
After=bitcoind.service
BindsTo=bitcoind.service
```

---

Wdyt of the above?

- It would solve the issue with BTC RPC Explorer not reading the new cookie file
- It would save some resources by automatically shutting down the Explorer when Bitcoin Core is not running
- It would simplify the shutting down of the node by simply having to terminate Bitcoin Core

Note: The same parametring would benefit Electrs.

**Comment by kristapsk** [comment]
One idea - could use `-startupnotify` of `bitcoind`, like we already do with setting permissions for RPC auth cookie (btw, probably will not be needed at some point anymore, if / when https://github.com/bitcoin/bitcoin/pull/26088 gets merged into Bitcoin Core), to `touch` some file somewhere (on `/tmp/`?) , and then use `inotifywait` to restart one or more other services, if they are present (see LND static channel backup script as an example for that).

**Comment by kristapsk** [comment]
Or we could even just watch `/home/bitcoin/.bitcoin/.cookie`, which is re-created on each `bitcoind` restart.

**Comment by VajraOfIndra** [comment]
> One idea [...]

Wdyt of the proposal above though? Seems simpler and feed two birds with one scone:
1. Work around the BTC RPC Explorer mishandling of the `.cookie` file
2. Save resources by stopping/starting the Explorer when Bitcoin Core stop/starts
